### ISSUE TITLE
Enums still not supported as of 3.6.2

### ISSUE THREAD

**Issue by jacekwilczynski** [squizlabs/PHP_CodeSniffer#3602]
**Describe the bug**
Despite https://github.com/squizlabs/PHP_CodeSniffer/pull/3478, there's still Generic.WhiteSpace.ScopeIndent errors when using enums.

**Code sample**
```php
<?php

declare(strict_types=1);

enum MyEnum
{
    case FIRST_OPTION;
    case SECOND_OPTION;
    case THIRD_OPTION;
}
```

**Custom ruleset**
```xml
<?xml version="1.0" encoding="UTF-8"?>

<ruleset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:noNamespaceSchemaLocation="vendor/squizlabs/php_codesniffer/phpcs.xsd">

    <arg name="basepath" value="."/>
    <arg name="cache" value="var/.phpcs-cache"/>
    <arg name="colors"/>
    <arg name="extensions" value="php"/>
    <arg value="p"/>
    <arg value="n"/>
    <arg value="s"/>

    <rule ref="PSR1"/>

    <rule ref="PSR12">
        <exclude name="PSR12.Operators.OperatorSpacing.NoSpaceBefore"/>
        <exclude name="PSR12.Operators.OperatorSpacing.NoSpaceAfter"/>
    </rule>

    <rule ref="Generic.Files.LineLength">
        <properties>
            <property name="lineLimit" value="120"/>
        </properties>
    </rule>
    <rule ref="Generic.Metrics.NestingLevel">
        <properties>
            <property name="nestingLevel" value="3"/>
            <property name="absoluteNestingLevel" value="3"/>
        </properties>
    </rule>
    <rule ref="Generic.PHP.ForbiddenFunctions"/>
    <rule ref="Generic.PHP.NoSilencedErrors"/>
    <rule ref="Generic.WhiteSpace.ScopeIndent"/>
    <rule ref="Generic.Formatting.SpaceAfterCast"/>

    <rule ref="Squiz.Classes.ValidClassName">
        <exclude name="Squiz.Classes.ValidClassName.NotCamelCaps"/>
    </rule>
</ruleset>
```

but a simple config like:
```xml
<?xml version="1.0" encoding="UTF-8"?>

<ruleset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:noNamespaceSchemaLocation="vendor/squizlabs/php_codesniffer/phpcs.xsd">

    <rule ref="Generic.WhiteSpace.ScopeIndent"/>
</ruleset>
```
produces an identical result.

**To reproduce**
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs test.php ...`
3. See error message displayed
```
  7 | ERROR | [x] Line indented incorrectly; expected 0 spaces, found 4 (Generic.WhiteSpace.ScopeIndent.IncorrectExact)
  8 | ERROR | [x] Line indented incorrectly; expected 0 spaces, found 4 (Generic.WhiteSpace.ScopeIndent.IncorrectExact)
  9 | ERROR | [x] Line indented incorrectly; expected 0 spaces, found 4 (Generic.WhiteSpace.ScopeIndent.IncorrectExact)
 10 | ERROR | [x] Line indented incorrectly; expected at least 4 spaces, found 0
    |       |     (Generic.WhiteSpace.ScopeIndent.Incorrect)
```

**Expected behavior**
No errors.

**Versions:**
 - OS: Debian GNU/Linux 11 (bullseye)
 - PHP: 8.1.6
 - PHPCS: 3.6.2
 - Standard: PSR1 + PSR12 + other rules (see config above)

**Additional context**
I am actually using the most recent version:
```
vendor/bin/phpcs --version
PHP_CodeSniffer version 3.6.2 (stable) by Squiz (http://www.squiz.net)
```

**Comment by jrfnl** [comment]
> Despite https://github.com/squizlabs/PHP_CodeSniffer/pull/3478, there's still Generic.WhiteSpace.ScopeIndent errors when using enums.

PHPCS 3.6.2 was released Dec 2021. [Syntax support for PHP 8.1](https://github.com/squizlabs/PHP_CodeSniffer/issues/3479) will be in version 3.7.0, which hasn't been released yet.

Please show some patience, we're all overworked and at the end of our energy after the ridiculous amount of changes for which supported needed to be added over the past few years. PHP 7.4, 8.0 and 8.1 have introduced more syntax changes and more impactful deprecations than all the PHP versions from PHP 5.0 - 7.3 combined.

**Comment by gsherwood** [comment]
I can confirm the issue exists in 3.6.2 but not in master, so this will be fixed in the 3.7.0 release.

The release milestone is listed on all issues and the milestone status lets you know if it's released and what's left to do before release. Here is a direct link for 3.7.0: https://github.com/squizlabs/PHP_CodeSniffer/milestone/28
